internal/pkg/manifest/validate.go

```diff
@@ -1721,6 +1738,52 @@ func validateDepsForEssentialContainers(deps map[string]containerDependency) err
 	return nil
 }
 
+func validateExposedPorts(opts validateExposedPortsOpts) error {
+	exposedPorts := make(map[uint16]string)
+
+	if opts.mainContainerPort != nil {
+		exposedPorts[aws.Uint16Value(opts.mainContainerPort)] = opts.mainContainerName
+	}
+
+	for name, sidecar := range opts.sidecarConfig {
+		if sidecar.Port == nil {
+			continue
+		}
+		sidecarPort, _, err := ParsePortMapping(sidecar.Port)
+		if err != nil {
+			return err
+		}
+
+		port, err := strconv.Atoi(aws.StringValue(sidecarPort))
+		if err != nil {
+			return err
+		}
+		if _, ok := exposedPorts[uint16(port)]; ok {
+			return &errContainersExposingSamePort{
+				firstContainer:  name,
+				secondContainer: exposedPorts[uint16(port)],
+				port:            aws.StringValue(sidecar.Port),
+			}
+		}
+		exposedPorts[uint16(port)] = name
+	}
+
+	// This condition takes care of the use case where target_container is set to x container and
+	// target_port exposing port 80 which is already exposed by container y.That means container x
+	// is trying to expose the port that is alreadt being exposed by container y, so error out.
+	if opts.targetPort != nil && opts.targetContainer != nil {
+		if exposedPorts[aws.Uint16Value(opts.targetPort)] != aws.StringValue(opts.targetContainer) {
+			return &errContainersExposingSamePort{
+				firstContainer:  aws.StringValue(opts.targetContainer),
+				secondContainer: exposedPorts[aws.Uint16Value(opts.targetPort)],
+				port:            strconv.Itoa(int(aws.Uint16Value(opts.targetPort))),
+			}
+		}
+	}
+
```

Comment:
do we need to handle these cases? or maybe they're handled somewhere else?
1. `target_port` not in `exposedPorts`
2. `target_container` not exposing any ports (or not existing?)

Reply:
Yes we are taking care of invalid target_port and target_container configuration in here.